src/corecel/random/data/RanluxppRngData.cc

```diff
@@ -0,0 +1,130 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+//---------------------------------------------------------------------------//
+/*!
+ * \file corecel/random/data/RanluxppRngData.cc
+ *
+ * Original source:
+ * https://github.com/apt-sim/AdePT/blob/master/include/AdePT/copcore/Ranluxpp.h
+ */
+//---------------------------------------------------------------------------//
+#include "RanluxppRngData.hh"
+
+#include <random>
+#include <vector>
+
+#include "corecel/data/Collection.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/random/data/RanluxppTypes.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize the state with the given seed.
+ */
+void initialize_state(RanluxppRngState& state,
+                      RanluxppUInt seed,
+                      HostCRef<RanluxppRngParamsData> const& params)
+{
+    // Skip 2 ** 96 states
+    RanluxppArray9 a_seed = celeritas::detail::compute_power_modulus(
+        params.kA_2048, RanluxppUInt(1) << 48);
+    a_seed = celeritas::detail::compute_power_modulus(a_seed,
+                                                      RanluxppUInt(1) << 48);
+
+    // Skip another s states.
+    a_seed = celeritas::detail::compute_power_modulus(a_seed, seed);
+    RanluxppArray9 lcg = {1, 0, 0, 0, 0, 0, 0, 0, 0};
+    lcg = celeritas::detail::compute_mod_multiply(a_seed, lcg);
+
+    // Set state and carry variable
+    state.state = celeritas::detail::to_ranlux(lcg, state.carry);
+    state.position = 0;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize Ranluxpp states with well-distributed random data.
+ *
+ * This generates pseudorandom, independent starting states for all data in
+ * all threads using a 32-bit MT19937 engine.
+ */
+void initialize_ranluxpp(Span<RanluxppRngState> states,
+                         HostCRef<RanluxppRngParamsData> const& params,
+                         StreamId stream)
+{
+    CELER_EXPECT(params);
+
+    // Generate well-distributed seed numbers, including StreamId so that each
+    // stream has different starting contribution
+    std::vector<std::seed_seq::result_type> host_seeds(1, params.seed);
+    if (stream != StreamId{0})
+    {
+        host_seeds.push_back(stream.get());
+    }
```

Comment:
BTW this was only needed for reproducibility with results before we had streams; in general we should just push the stream ID.